### PR TITLE
[native] Add conversion from Presto protocol WindowNode to Velox WindowNode

### PR DESCRIPTION
**Pull request by aditi-pandit**
Test plan - TestHiveWindowQueries unit test

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-native-execution/src/test/java/com/facebook/presto/nativeworker/TestHiveWindowQueries.java [hunk 1]
```diff
@@ -23,9 +23,9 @@ public TestHiveWindowQueries()
         super(true);
     }
 
-    @Test (enabled = false)
+    @Test
     public void testWindow()
     {
-        assertQueryFails("SELECT clerk, orderdate, orderkey, totalprice, rank() OVER (PARTITION BY clerk ORDER BY totalprice) AS rnk FROM orders ORDER BY clerk, rnk", ".*Unknown plan node type com.facebook.presto.sql.planner.plan.WindowNode.*");
+        assertQuery("SELECT clerk, orderdate, orderkey, totalprice, rank() OVER (PARTITION BY clerk ORDER BY totalprice) AS rnk FROM orders ORDER BY clerk, rnk");
```
Comment (mbasmanova) [inline presto-native-execution/src/test/java/com/facebook/presto/nativeworker/TestHiveWindowQueries.java:29]: Would you add queries for all currently supported functions?

File: presto-native-execution/presto_cpp/main/types/PrestoToVeloxExpr.cpp [hunk 2]
```diff
@@ -81,7 +81,7 @@ std::string mapScalarFunction(const std::string& name) {
   return lowerCaseName;
 }
 
-std::string mapAggregateFunctionName(const std::string& name) {
+std::string mapAggregateAndWindowFunction(const std::string& name) {
```
Comment (mbasmanova) [inline presto-native-execution/presto_cpp/main/types/PrestoToVeloxExpr.cpp:84]: nit: mapAggregateAndWindowFunction -> mapAggregateOrWindowFunction

**Review by mbasmanova — APPROVED** [review]
@aditi-pandit This is nice. Looks good to me % one comment for the test.

File: presto-native-execution/src/test/java/com/facebook/presto/nativeworker/TestHiveWindowQueries.java (hunk 1 above)
Comment (aditi-pandit) [reply presto-native-execution/src/test/java/com/facebook/presto/nativeworker/TestHiveWindowQueries.java:30]: I have added one query per window function supported now. Though I have plans to add more exhaustive tests in the future.

**Pull request merged**